cmd/sonobuoy/app/kubeconfig.go

```diff
@@ -37,6 +37,14 @@ func (c *Kubeconfig) Set(str string) error {
 
 // Get returns a rest Config, possibly based on a provided config
 func (c *Kubeconfig) Get() (*rest.Config, error) {
+
+	// detect if in-cluster configuration makes sense
```

Comment:
This is already handled by https://github.com/heptio/sonobuoy/blob/master/cmd/sonobuoy/app/kubeconfig.go#L46

That call to ClientConfig will return in cluster config if it needs to.